src/Forms/FormField.php

```diff
@@ -1222,32 +1234,30 @@ public function Type()
     }
 
     /**
-     * Utility method to call an extension hook which allows the result of validate() calls to be adjusted
-     *
-     * @param bool $result
-     * @param Validator $validator
-     * @return bool
-     * @deprecated 5.4.0 Use extend() directly instead
+     * Determines whether the field is valid or not based on its value
      */
-    protected function extendValidationResult(bool $result, Validator $validator): bool
+    public function validate(): ValidationResult
     {
-        Deprecation::notice('5.4.0', 'Use extend() directly instead');
-        $this->extend('updateValidationResult', $result, $validator);
+        if (count(func_get_args()) !== 0) {
+            // In prior versions, FormField::validate() had a method signature of
+            // validate(Validator $validator): bool
+            // Old code may still call this method this without throwing an error, however the
+            // return type is now ValidationResult which may be mistakenly interpreted as truthy
+            // While we cannot stop people from interpreting the return value as a boolean, we can
+            // at least throw an exception so they might notice something is wrong.
+            $message = 'validate() does not accept any arguments.'
+            . ' Note that the return type is ValidationResult, not boolean.';
+            throw new InvalidArgumentException($message);
+        }
```

Comment:
This feels like we're doing something in CMS 6 that should be covered by a deprecation in 5 with relevant changelog information. I won't block merging for this because I can't see us agreeing about it, but IMO this is adding tech debt.

Reply:
Removed. Have added CMS 5 deprecation notice, 5.4.0 changelog, and 6.0.0 changelog notes